tests/integration/test_charm_metrics.py

```diff
@@ -0,0 +1,66 @@
+#  Copyright 2023 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Integration tests for metrics."""
+import json
+import logging
+
+from juju.application import Application
+from juju.model import Model
+from juju.unit import Unit
+
+from metrics import METRICS_LOG_PATH
+from tests.integration.helpers import wait_till_num_of_runners
+from tests.status_name import ACTIVE_STATUS_NAME
+
+
+async def _create_runner(app: Application, model: Model) -> None:
+    """Let the charm create a runner.
+
+    Args:
+        app: The GitHub Runner Charm app to create the runner for.
+        model: The marchine charm model.
+    """
+    await app.set_config({"virtual-machines": "1"})
+    unit = app.units[0]
+    action = await unit.run_action("reconcile-runners")
+    await action.wait()
+    await model.wait_for_idle(apps=[app.name], status=ACTIVE_STATUS_NAME)
+    await wait_till_num_of_runners(unit, 1)
```

Comment:
Is there no fixture or helper function that does this already for another test? Just checking so we don't re-do something that is already done and I think I remember other tests that have runners in them

Reply:
I'm guessing that it is unique to this test since the relation is required before the runner is created. Perhaps see if this function could be used in the other cases when runners are being created?

Reply:
I moved it to `helpers.py` and reused it in a few places.